### DISCUSSION TITLE
Macos and non-Nixos Linux working configuration?

### DISCUSSION THREAD

**Discussion by rhodee** [juspay/nix-dev-home#99] [Q&A] [answered]
I had a question regarding how to setup multiple machines. I am looking to manage one macos and one Pop!OS machine using nixos-unified. I can get each working independently, but I would like to create a config which can setup home-manager for both.

My current config follows the patterns laid out in the templates [here](https://nixos-unified.org/examples.html):

```sh
.
├── configurations
│   ├── darwin
│   │   └── mac.nix
│   └── home
│       └── rhodee.nix
├── flake.lock
├── flake.nix
└── modules
    ├── darwin
    │   └── default.nix
    ├── flake-parts
    │   ├── activate-home.nix 
    │   ├── devshell.nix
    │   └── toplevel.nix
    └── home
        ├── all
        ├── default.nix
        ├── editor
        ├── fonts.nix
        ├── gc.nix
        ├── git
        ├── linux.nix
        ├── macos.nix
        ├── nix-index.nix
        └── packages.nix
```

On Linux I need to have `modules/flake-parts/activate-home.nix` present otherwise `nix run` fails. However when I have this file in the same directory on my Mac, the build result passes but certain programs stop working and I see logs I would have expected only in Linux environments.

How can I structure my configuration such that it works for both OS environments in the same repo? Is it possible to only trigger `activate-home.nix` only on linux instead of in all environments?

**Comment by srid** [comment, marked as answer]
The quick answer for now is:

Delete `activate-home.nix`, and run `nix run . $USER@` on PopOS while running `nix run` on macOS. 

But yes, a proper generic solution here is needed so that `nix run` works for all platforms. Let me think about this. I suppose we can have the [activate script](https://nixos-unified.org/activate.html) inspect the system type and act accordingly.

**Reply by rhodee** [reply]
Thanks @srid. This projet and its previous iteration have been great for my productivity, thank you. I will make the update in my dotfiles.

**Reply by MikaelElkiaer** [reply]
I assume the broken link refers to https://github.com/srid/nixos-unified/blob/1f8ab18330354d2305a0d793da58a6ef83e2857c/activate/activate.nu.

Has anyone worked on enabling what @srid describes?
I mean, updating the script so that `nix run` works even for home-manager only set-ups.
